rellic/AST/ExprCombine.cpp

```diff
@@ -95,12 +95,13 @@ class DerefAddrOfRule : public InferenceRule {
     auto deref = clang::cast<clang::UnaryOperator>(stmt);
     CHECK(deref == match)
         << "Substituted UnaryOperator is not the matched UnaryOperator!";
-    auto addr_of = clang::cast<clang::UnaryOperator>(deref->getSubExpr());
+    auto subexpr = deref->getSubExpr()->IgnoreParens();
```

Comment:
Can types be an issue here? E.g. `*(uint8_t *)&int_val;`?

Reply:
Do you think it'd be reasonable, where the types don't match, to substitute with the appropriate C++ case type, i.e. `static_cast`, `reinterpret_cast`, or even fall back to a `memcpy` in some cases, and if the desired output is C, then those more explicit casting forms can be changed later?

Reply:
Type casts will be an issue here in the sense that the matcher of this rule will not match `*(uint8_t *)&int_val` since there's an extra C-style cast expression that is not ignored.